polkadot-protocol/introduction/glossary.md

```diff
@@ -0,0 +1,384 @@
+
+## Keystore
+
+A subsystem for managing keys for the purpose of producing new blocks.
+
+## Kusama
+
+[Kusama](https://kusama.network/) is a Substrate-based [blockchain](#blockchain) that implements a design similar to the [Polkadot network](#polkadot-network).
+Kusama is a [canary](https://en.wiktionary.org/wiki/canary_in_a_coal_mine) network and is referred to as [Polkadot's "wild cousin"](https://wiki.polkadot.network/docs/learn-comparisons-kusama).
+As a canary network, Kusama is expected to be more stable than a test network like [Westend](#westend), but not as stable as a production network like [Polkadot](#polkadot).
+
+As a canary network, Kusama is controlled by its network participants and is intended to be stable enough to encourage meaningful experimentation.
+
+## Libp2p
+
+A peer-to-peer networking stack that allows use of many transport mechanisms, including WebSockets (usable in a web browser).
+Substrate uses the [Rust implementation](https://github.com/libp2p/rust-libp2p) of the `libp2p` networking stack.
+
+## Light Client
+
+A type of blockchain [node](#node) that does not store the [chain state](#state) or produce blocks.
+A light client is capable of verifying [cryptographic primitives](#cryptographic-primitives) and exposes a [remote procedure call (RPC)](https://en.wikipedia.org/wiki/Remote_procedure_call) server that allows blockchain users to interact with the blockchain network.
+
+## Metadata
+
+Data that provides information about one or more aspects of a system.
+The metadata that exposes information about a Substrate [blockchain](#blockchain) enables you to interact with that system.
+
+## Node
+
+A running instance of a blockchain client.
+Each node is part of the [peer-to-peer](https://en.wikipedia.org/wiki/Peer-to-peer) network that allows blockchain participants to interact with one another.
+Substrate nodes can fill a number of roles in a blockchain network.
+For example, the nodes that produce blocks fulfill the [validator](#validator) role for the blockchain.
+Nodes that run [light-clients](#light-client) facilitate scalable interactions in resource-constrained environments like [user interfaces](https://github.com/paritytech/substrate-connect) or embedded devices.
+
+## Nominated Proof of Stake (NPoS)
+
+A method for determining [validators](#validator) or _[authorities](#authority)_ based on a willingness to commit their stake to the proper functioning of one or more block producing nodes.
+
+## Oracle
+
+In a blockchain network, an oracle is a mechanism for connecting the blockchain to a non-blockchain data source.
+Oracles enable the blockchain to access and act upon information from existing data sources and incorporate data from non-blockchain systems and services.
+
+## Origin
+
+A [FRAME](#frame) primitive that identifies the source of a [dispatched](#dispatchable) function call into the [runtime](#runtime).
+The FRAME `system` pallet defines three built-in [origins](#origin).
```

Comment:
```suggestion
The FRAME System pallet defines three built-in [origins](#origin).
```

When casually referring to a pallet, we do not need to wrap it in back ticks. Instead captialize the name of the pallet